docs/demos/first_steps.md

```diff
@@ -0,0 +1,299 @@
+
+
+## Download the OSM command-line tool
+
+The `osm` command-line tool contains everything needed to install and configure Open Service Mesh.
+The binary is available on the [OSM GitHub releases page](https://github.com/openservicemesh/osm/releases/).
+
+### For GNU/Linux or macOS
+Download and unzip the 64-bit [GNU/Linux](https://github.com/openservicemesh/osm/releases/download/v0.8.0-rc.2/osm-v0.8.0-rc.2-linux-amd64.tar.gz)
+or
+[macOS](https://github.com/openservicemesh/osm/releases/download/v0.8.0-rc.2/osm-v0.8.0-rc.2-darwin-amd64.tar.gz)
+binary of OSM v0.8.0-rc.2:
+```bash
+system=$(uname -s | tr '[:upper:]' '[:lower:]')
+release=v0.8.0-rc.2
+curl -L https://github.com/openservicemesh/osm/releases/download/${release}/osm-${release}-${system}-amd64.tar.gz | tar -vxzf -
+./${system}-amd64/osm version
+```
+
+### For Windows
+Download and unzip the Windows OSM v0.8.0-rc.2 binary via PowerShell:
+```powershell
+wget  https://github.com/openservicemesh/osm/releases/download/v0.8.0-rc.2/osm-v0.8.0-rc.2-windows-amd64.zip -o osm.zip
+unzip osm.zip
+.\windows-amd64\osm.exe version
+```
+
+### Compile from source
+The `osm` CLI can be compiled from source using [this guide](../../install/_index.md).
+
+
+## Installing OSM on Kubernetes
+
+With the `osm` binary downloaded and unzipped, we are ready to install Open Service Mesh on a Kubernetes cluster:
+
+```bash
+osm install
+```
+
+> Note: This document assumes you have already installed credentials for a Kubernetes cluster in ~/.kube/config and `kubectl cluster-info` executes successfully.
+
+This installed OSM Controller in the `osm-system` namespace.
+
+## Deploy Demo Applications
+In this section we will deploy 2 pods in 2 different namespaces:
+- `bookbuyer` - a client, which makes HTTP requests to a `bookstore`
+- `bookstore` - a server, which responds to HTTP requests from `bookbuyer`
+
+
+### Create the Namespaces
+Each application in this demo will reside in a separate namespaces. Create the namespaces with:
+```bash
+kubectl create namespace bookbuyer
+kubectl create namespace bookstore
+```
+
+### Join the new namespaces to the service mesh
+The following `osm` CLI command joins the namespaces to the service mesh:
+
+`//TODO(draychev): link to relevant documentation on how the JOIN command works [#2904]`
+
+```bash
+osm namespace add bookbuyer
+osm namespace add bookstore
+```
+
+With this command each one of the four namespaces will be:
+  1. labelled with `openservicemesh.io/monitored-by: osm`
+  2. annotated with `openservicemesh.io/sidecar-injection: enabled`
+
+The OSM Controller will notice the new label and annotation on
+these namespaces and will begin injecting **new** pods with Envoy sidecars.
+
+### Create Pods
+
+Create the `bookbuyer` service account:
+```bash
+kubectl apply -f - <<EOF
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: bookbuyer
+  namespace: bookbuyer
+EOF
```

Comment:
`<<EOF` is specific to unix shells, is it noted somewhere in the guide these commands are meant to be run in such an environment?

Reply:
One of the 3 requirements for this demo is `  - a workstation capable of executing [Bash](https://en.wikipedia.org/wiki/Bash_(Unix_shell)) scripts`

heredocs are extremely nice for docs like that